pkg/virt-operator/resource/apply/core_test.go

```diff
@@ -123,6 +135,120 @@ var _ = Describe("Apply", func() {
 		})
 	})
 
+	Context("should reconcile service account", func() {
+
+		newServiceAccount := func() *corev1.ServiceAccount {
+			return &corev1.ServiceAccount{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "ServiceAccount",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "namespace",
+					Name:      "name",
+				},
+			}
+		}
+
+		var clientset *kubecli.MockKubevirtClient
+		var ctrl *gomock.Controller
+		var coreclientset *fake.Clientset
+		var expectations *util.Expectations
+		var kv *v1.KubeVirt
+		var stores util.Stores
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			kvInterface := kubecli.NewMockKubeVirtInterface(ctrl)
+			coreclientset = fake.NewSimpleClientset()
+
+			coreclientset.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				Expect(action).To(BeNil())
+				return true, nil, nil
+			})
+
+			stores = util.Stores{}
+			stores.ServiceAccountCache = cache.NewStore(cache.DeletionHandlingMetaNamespaceKeyFunc)
+			stores.InstallStrategyConfigMapCache = cache.NewStore(cache.MetaNamespaceKeyFunc)
+
+			expectations = &util.Expectations{}
+
+			clientset = kubecli.NewMockKubevirtClient(ctrl)
+			clientset.EXPECT().KubeVirt(Namespace).Return(kvInterface).AnyTimes()
+			clientset.EXPECT().CoreV1().Return(coreclientset.CoreV1()).AnyTimes()
+
+			kv = &v1.KubeVirt{}
+		})
+
+		AfterEach(func() {
+			ctrl.Finish()
+		})
+
+		It("should not patch ServiceAccount on sync when they are equal", func() {
+
+			pr := newServiceAccount()
+
+			version, imageRegistry, id := getTargetVersionRegistryID(kv)
+			injectOperatorMetadata(kv, &pr.ObjectMeta, version, imageRegistry, id, true)
+
+			stores.ServiceAccountCache.Add(pr)
+
+			r := &Reconciler{
+				kv:           kv,
+				stores:       stores,
+				clientset:    clientset,
+				expectations: expectations,
+			}
+
+			Expect(r.createOrUpdateServiceAccount(pr)).To(BeNil())
+		})
+
+		It("should patch ServiceAccount on sync when they are equal", func() {
```

Comment:
"when they are not equal"?

I assume that's what you meant since an annotation was added.